### PR TITLE
Changes for 0.25

### PR DESCRIPTION
**Pull request by wilzbach**
Be careful when reviewing. There were a few things that felt awkward and we should have better language capabilities for. Additionally, I had to workaround a few bugs.

### CODE REVIEW

File: src/cli/login.story [hunk 1]
```diff
@@ -38,23 +38,23 @@ http server as client
...
-        emails = http fetch url: "https://api.github.com/user/emails" headers: headers
+        emails = (http fetch url: "https://api.github.com/user/emails" headers: headers) to List[Map[string,string]]
         primary_email = emails[0]["email"]
 
         # Insert into postgres.
-        service_id = user["id"] as string
-        creds_raw = psql exec query: "select create_owner_by_login(%(service)s, %(service_id)s, %(username)s, %(name)s, %(email)s, %(oauth_token)s) as data" data: {"service": "github", "service_id": service_id, "username": user["login"], "name": user["name"], "email": primary_email, "oauth_token": token}
-        creds = creds_raw[0]["data"]
+        service_id = user["id"] to string
+        creds_raw = (psql exec query: "select create_owner_by_login(%(service)s, %(service_id)s, %(username)s, %(name)s, %(email)s, %(oauth_token)s) as data" data: {"service": "github", "service_id": service_id, "username": user["login"], "name": user["name"], "email": primary_email, "oauth_token": token}) to List[Map[string,any]]
```
Comment (wilzbach) [inline src/cli/login.story]: We could use `psql select` here to directly get typing, but that one only supports `Map[string,string]`.

File: src/handlers/github_webhooks.story [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
-    if req.headers["X-Github-Event"] == "release" and req.body["release"]["draft"] == false and req.body["action"] == "published"
-      release = req.body["release"]
+    xEvent = req.headers["X-Github-Event"] to string
+    if xEvent == "release"
+        return
+
+    body = req.body to Map[string, any]
+    action = body["action"] to string
+    release = body["action"] to Map[string,any]
+    if (release["draft"] to boolean) and action == "published"
```
Comment (wilzbach) [inline src/handlers/github_webhooks.story]: workaround for https://github.com/storyscript/storyscript/issues/1296

File: src/cli/login.story [hunk 3]
```diff
@@ -38,23 +38,23 @@ http server as client
         # Get the oauth_token.
         body = {"client_id": app.secrets.github_client_id, "client_secret": app.secrets.github_client_secret, "code": code, "state": state}
         headers = {"Content-Type": "application/json; charset=utf-8", "Accept": "application/json"}
-        gh_response = http fetch url: "https://github.com/login/oauth/access_token" method: "post" body: body headers: headers
+        gh_response = (http fetch url: "https://github.com/login/oauth/access_token" method: "post" body: body headers: headers) to Map[string,string]
```
Comment (wilzbach) [inline src/cli/login.story]: I think we should introduce a `fetchJSON` for better typing.

File: src/cli/login.story [hunk 4]
```diff
@@ -71,6 +71,5 @@ http server as client
             request write content: "null"
             return
 
-        user_data = json parse content: user_data
-        request set_header key: "Content-Type" value: "application/json; charset=utf-8"
-        request write content: (json stringify content: user_data)
+        user_response = json parse content: user_data
```
Comment (wilzbach) [inline src/cli/login.story]: Not sure why parse and then serialize again.

File: src/handlers/github_webhooks.story [hunk 5]
```diff
@@ -1,9 +1,16 @@
...
       tag_name = release["tag_name"]
-      full_name = req.body["repository"]["full_name"]
+      repo = body["repository"] to Map[string,string]
+      full_name = repo["full_name"]
```
Comment (wilzbach) [inline src/handlers/github_webhooks.story]: workaround for https://github.com/storyscript/storyscript/issues/1297

File: src/handlers/jira_sync.story [hunk 6]
```diff
@@ -18,16 +18,20 @@ function mapGHUsernameToJiraAccountId username: string returns string
...
 
 function build_jira_request_body body:any returns any
-    issue_title = body["issue"]["title"]
-    issue_title_prefix = body["repository"]["full_name"]
-    reporterAccountId = mapGHUsernameToJiraAccountId(username: body["issue"]["user"]["login"])
+    jiraBody = body to Map[string,any]
+    issue = jiraBody["issue"] to Map[string,string]
+    issue_title = issue["title"]
+    issue_title_prefix = issue["full_name"]
+    issue_nested = jiraBody["issue"] to Map[string,Map[string,string]]
```
Comment (wilzbach) [inline src/handlers/jira_sync.story]: workaround for https://github.com/storyscript/storyscript/issues/1297

File: src/handlers/jira_sync.story [hunk 7]
```diff
@@ -65,31 +69,31 @@ function get_auth_header_value returns string
...
 
-    return out as string
+    return out to string
 
 
 function create_jira_issue body: Map[string, any] returns string
     headers = {"Authorization": get_auth_header_value(), "Content-Type": "application/json"}
     jira_payload = build_jira_request_body(body:body)
-    res = http fetch method: "post" url: "https://storyscript.atlassian.net/rest/api/3/issue" headers: headers body: jira_payload
+    res = (http fetch method: "post" url: "https://storyscript.atlassian.net/rest/api/3/issue" headers: headers body: jira_payload) to Map[string,string]
```
Comment (wilzbach) [inline src/handlers/jira_sync.story]: `fetchJSON` would help.

File: src/handlers/jira_sync.story [hunk 8]
```diff
@@ -147,31 +153,36 @@ function assignJiraIssue jiraIssueId: string body: Map[string, any]
...
 
-    if req.body["action"] == "opened"
-        if is_author_a_team_member(gh_payload: req.body)
-            create_jira_issue(body: req.body)
-    else if req.body["action"] == "assigned"
-        if not is_author_a_team_member(gh_payload: req.body)
+    body = req.body to Map[string,any]
+    issue = body["issue"] to Map[string,string]
+    issue_id = issue["id"] to int
+    action = body["action"] to string
```
Comment (wilzbach) [inline src/handlers/jira_sync.story]: Not very elegant.

File: src/analytics/router.story [hunk 9]
```diff
@@ -13,5 +13,6 @@ http server as server
     clevertap push profile: profile identity: id
 
   when server listen path: "/track/ghost/subscriber/added" method: "post" as req
-    email = req.body["subscribers"][0]["email"]
+    subscribers = req.body["subscribers"] to List[Map[string,string]]
```
Comment (wilzbach) [inline src/analytics/router.story]: A `listenJSON` could help.

File: src/cli/login.story [hunk 10]
```diff
@@ -38,23 +38,23 @@ http server as client
         # Get the oauth_token.
         body = {"client_id": app.secrets.github_client_id, "client_secret": app.secrets.github_client_secret, "code": code, "state": state}
         headers = {"Content-Type": "application/json; charset=utf-8", "Accept": "application/json"}
-        gh_response = http fetch url: "https://github.com/login/oauth/access_token" method: "post" body: body headers: headers
+        gh_response = (http fetch url: "https://github.com/login/oauth/access_token" method: "post" body: body headers: headers) to Map[string,string]
         token = gh_response["access_token"]
 
         headers = {"Authorization": "bearer {token}"}
-        user = http fetch url: "https://api.github.com/user" headers: headers
+        user = (http fetch url: "https://api.github.com/user" headers: headers) to Map[string,string]
```
Comment (wilzbach) [inline src/cli/login.story]: Another candidate for `fetchJson`

File: src/cli/login.story (hunk 4 above)
Comment (judepereira) [reply src/cli/login.story]: There used to be an if condition. Anyway, we don't need to do so anymore.

Comment (judepereira) [reply src/cli/login.story]: Why'd you remove the Content-Type header?

Comment (wilzbach) [reply src/cli/login.story]: Because `writeJSON` is a thing: https://github.com/storyscript/http/pull/55

**Pull request merged**